backend/app/groups/routes.py

```
@@ -145,3 +155,64 @@
+    current_user: dict = Depends(get_current_user),
+):
+
+    await group_service.ensure_user_in_group(group_id, current_user["_id"])
+
+    try:
+        urls = await storage_service.upload_image_workflow(
+            file=file, folder="groups", entity_id=group_id
+        )
+
+    except ValueError as ve:
+        raise HTTPException(status_code=400, detail=str(ve))
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)